pkg/controller/clusterstorage/clusterstorage_controller_test.go

```diff
@@ -93,3 +98,85 @@ func TestSetStatusProgressing(t *testing.T) {
 		})
 	}
 }
+
+func TestReconcile(t *testing.T) {
+	volumeExpansion := false
+	newStorageClass := resourceread.ReadStorageClassV1OrDie(generated.MustAsset("assets/aws.yaml"))
```

Comment:
nit: IMO the test could be more representative if you make `newStorageClass` a field in the anonymous struct below.

That way we could add more test cases with different variations, plus everything would be more deterministic (e.g., currently the test changes if we change the `assets/aws.yaml` file).

Reply:
Updated the test to fit in with the typical method of testing, and included a helper method to obtain the fake storage class.